Integrations/integration-Cisco-umbrella.yml

```diff
@@ -31,495 +31,1834 @@ configuration:
+        if params_dict:
+            req_params.update(params_dict)
+        url = BASE_URL + api_endpoint
+        LOG('running %s request with url=%s\theaders=%s\nparams=%s\tdata=%s' % (method, url, DEFAULT_HEADERS, json.dumps(req_params), data_list))
+        try:
+            res = requests.request(
+                method,
+                url,
+                verify=USE_SSL,
+                params=req_params,
+                headers=DEFAULT_HEADERS,
+                data=data_list
+            )
+
+            res.raise_for_status()
+            return res.json()
+
+        except Exception, e:
+            LOG(e)
+            raise
+
+
+    def string_to_table_header(string):
+        # Checks if string, change underscore to space, capitalize every word. example: "one_two" to "One Two" (title() could work too)
+        if type(string) in STRING_TYPES:
+            return " ".join(word.capitalize() for word in string.replace("_"," ").split())
+        else:
+            return_error('The key is not a string: {}'.format(string))
+
+
+    def string_to_context_key(string):
+        # Checks if string, removes underscores, capitalize every word. example: "one_two" to "OneTwo" (title() could work too)
+        if type(string) in STRING_TYPES:
+            return "".join(word.capitalize() for word in string.split('_'))
+        else:
+            return_error('The key is not a string: {}'.format(string))
+
+
+    def date_to_ts(date):
+        # this helper function tries to parse a date time string according to a specific format
+        # if it fails, it will just output the original value
+        try:
+            ts = datetime.strptime(date[0:16],'%Y-%m-%dT%H:%M')
+        except ValueError:
+            pass
+        else:
+            if date[19]=='+':
+                ts+=timedelta(hours=int(date[20:22]),minutes=int(date[22:24]))
+            elif date[19]=='-':
+                ts-=timedelta(hours=int(date[20:22]),minutes=int(date[22:24]))
+            ts = time.mktime(ts.timetuple())
+            return str(int(ts) * 1000)
+        return date
+
+
+    def ts_to_date(ts):
+        if ts:
+        # Gets a timestamp (either str or int, either in seconds or milliseconds) and converts it to a date.
+            ts = str(ts)
+            if len(ts) > 10:
+                ts = ts[:10]
+            ts = int(ts)
+            return datetime.utcfromtimestamp(ts).strftime('%Y-%m-%dT%H:%M:%S')
+        return ts
+
+    def securerank_to_dbotscore(sr):
+        # converts cisco umbrella score to dbotscore
+        DBotScore = 0
+        if sr > 0 and sr <= 100:
+            DBotScore = 1
+        elif sr < 0 and sr > -100:
+            DBotScore = 2
+        elif sr == -100:
+            DBotScore = 3
+        return DBotScore
+
+
+    ''' INTERNAL FUNCTIONS '''
+
+    def get_co_occurences(domain):
+        # Build & Send request
+        endpoint_url = '/recommendations/name/' + domain + '.json'
```

Comment:
I think it would be better to make a CONST out of it, use `'/recommendations/name/{0}.json'`
and then here you can use `.format(domain)`

Reply:
You mean CONST for endpoints basically? not sure @anara123 would approve